test/extensions/filters/http/compressor/compressor_filter_test.cc

```diff
@@ -430,6 +430,7 @@ TEST_F(CompressorFilterTest, EmptyResponse) {
 
 // Verify removeAcceptEncoding header.
 TEST_F(CompressorFilterTest, RemoveAcceptEncodingHeader) {
```

Comment:
I'm open to feedback on making this test less copy-pastey.  I wasn't able to replicate [this similar situation](https://github.com/envoyproxy/envoy/blob/33e7d80206c099fbff898a3402758757cab49436/test/extensions/filters/http/compressor/compressor_filter_test.cc#L188-L211) because it's relying on the top-level filter flag being a runtime flag, which is not the case with `remove_accept_encoding_header`.